File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/projection/isproductactive/IsProductActive.java
package cloud.ambar.creditcard.enrollment.projection.isproductactive;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.web.context.annotation.RequestScope;

@Service
@RequestScope
@RequiredArgsConstructor
public class IsProductActive {
    private final ProductActiveStatusRepository productActiveStatusRepository;

    public boolean isProductActive(final String productId) {
        return productActiveStatusRepository.isThereAnActiveProductWithId(productId);
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Projection/IsProductActive/IsProductActive.cs
namespace CreditCardEnrollment.CreditCard.Enrollment.Projection.IsProductActive;

public class IsProductActive {
    private readonly ProductActiveStatusRepository _productActiveStatusRepository;

    public IsProductActive(ProductActiveStatusRepository productActiveStatusRepository) {
        _productActiveStatusRepository = productActiveStatusRepository;
    }

    public bool IsProductActiveById(string productId) {
        return _productActiveStatusRepository.IsThereAnActiveProductWithId(productId);
    }
}